src/xml/gffcreator.cpp

```diff
@@ -0,0 +1,57 @@
+/* xoreos-tools - Tools to help with xoreos development
+ *
+ * xoreos-tools is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos-tools is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos-tools is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos-tools. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  Creates GFFs out of XML files.
+ */
+
+#include <src/common/error.h>
+#include "src/xml/xmlparser.h"
+
+#include "gffcreator.h"
+#include "gff3creator.h"
+
+namespace XML {
+
+void GFFCreator::create(Common::WriteStream &output, Common::ReadStream &input, const Common::UString &inputFileName) {
+	XMLParser xml(input, true, inputFileName);
+
+	const XMLNode &xmlRoot = xml.getRoot();
+
+	uint32 typeId;
+	Common::UString type = xmlRoot.getProperty("type");
+	for (size_t i = 0; i < 4; ++i) {
+		if (type.size() > i)
+			reinterpret_cast<char*>(&typeId)[3 - i] = *type.getPosition(i);
+		else
+			reinterpret_cast<char*>(&typeId)[3 - i] = ' ';
+	}
```

Comment:
That's...pretty icky. Can you use `MKTAG()` instead?

You don't even need the loop, you can just pad the string with 4 spaces beforehand. 

Like
```
const Common::UString type = xmlRoot.getProperty("type") + "    ";
const uint32 typeID = MKTAG(*type.getPosition(0), ...
```

(Theoretically, you could even use Common::UString::at() from a6f1c2bdb53785da394dc9dbdb146245eee72925)